tests/testthat/test-derive_param_response.R

```diff
@@ -83,10 +82,68 @@ test_that("Test that response is derived properly", {
     ) %>%
     arrange(USUBJID, PARAMCD, ADT)
 
-
   expect_dfs_equal(
     actual_output,
     expected_output,
     keys = c("USUBJID", "PARAMCD", "ADT")
   )
 })
+
+## Test 2: No source_pd ----
+test_that("derive_param_response Test 2: No source_pd", {
+  rsp <- adrs1 %>%
+    filter(PARAMCD == "OVR" & AVALC %in% c("CR", "PR")) %>%
+    left_join(adrs1 %>%
+      filter(PARAMCD == "PD") %>%
+      select(STUDYID, USUBJID, PDDT = ADT),
+    by = c("STUDYID", "USUBJID")
+    ) %>%
+    filter(ADT <= PDDT | is.na(PDDT)) %>%
+    arrange(USUBJID, PARAMCD, ADT) %>%
+    group_by(USUBJID, PARAMCD) %>%
+    filter(row_number() == 1) %>%
+    mutate(avalc = 1)
+
+  new_obs <- adsl %>%
+    left_join(rsp, by = c("STUDYID", "USUBJID")) %>%
+    mutate(
+      AVALC = if_else(avalc == 1, "Y", "N", "N"),
+      AVAL = if_else(avalc == 1, 1, 0, 0),
+      PARAMCD = "RSP", PARAM = "Response by investigator"
+    ) %>%
+    select(-avalc, -PDDT)
```

Comment:
Please do not derive the expected dataset. (See the note in https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#set-up-the-test-script.)

Reply:
@bundfussr are you ok to make this fix on Samia's behalf as she's tied up with project work?

Reply:
I can make the change then, @rossfarrugia @bundfussr, I just need to get my head round the requirements a little first.  

Realistically though I wont get a chance till tomorrow, if this is ok?

Reply:
@sgorm123 , it is OK from my side that you update both tests.